### ISSUE TITLE
The page is still cached (by django) for anonymous user after publishing new content on the page

### ISSUE THREAD

**Issue by gordeychuk** [django-cms/django-cms#6975]
## Description
The page is still cached for anonymous user after publishing new content on the page. Everything works good for logged-in users (because caching is not active for logged-in users as far as I understand). I've investigated the cache DB and cache middlewares (`django.middleware.cache.UpdateCacheMiddleware` and `django.middleware.cache.FetchFromCacheMiddleware`) source code and found that basically it uses `views.decorators.cache.cache_header...` and `views.decorators.cache.cache_page...` cache keys to get/set the page cache, but django CMS is not updating them on publishing. Django CMS is setting its own cache keys which are not related to the ones mentioned above.

The issue can be avoided by setting `CMS_PLUGIN_CACHE=False` in the settings. I still want to use the cache, so I've implemented a signal receiver for the page publishing signal for removing the related cache key (see additional info). 

## Steps to reproduce
1. Configure cache (add required middlewares and set CMS_CACHE_DURATIONS)
2. Configure database cache using: 
```python
CACHES = {'default': {'BACKEND': 'django.core.cache.backends.db.DatabaseCache', 'LOCATION': 'django_dbcache'}}
```
3. Run server
4. **As logged-in user:** create a page with simple content (like Text Plugin)
5. **As anonymous user:** open the page and check that content is valid
6. **As logged-in user:** update the content and publish the page
7. **As anonymous user:** reload the page 

## Expected behaviour
The updated content should be displayed for the anonymous user.

## Actual behaviour
The cached (old) content is displayed for the anonymous user. 

## Additional information

django-cms: 3.8.0
Django: 3.1.6
OS: OS X 11.1 (but also reproducible on Windows 10)
Browsers: Google Chrome 88.0.4324.146, Firefox 85.0.2
Also I've attached the example project where you can see the issue: [django_cms_caching_issue.zip](https://github.com/django-cms/django-cms/files/5959204/django_cms_caching_issue.zip)

The workaround which I used to keep cache working:
Add a signal receiver to any of the existing apps or create a new app. 
The receiver code:

```python
import logging

from cms.admin.pageadmin import PageAdmin
from django.core.cache import cache
from cms import signals
from django.utils.cache import _generate_cache_header_key

logger = logging.getLogger(__name__)


def delete_cache_key(sender, obj, operation, request,  **kwargs):
    try:
        if operation == 'publish_page_translation':
            # In order to get the same cache_key as the user would receive we need to create a WSGIRequest
            # object with empty query string and with page path
            request_custom = request
            request_custom.path = obj.get_absolute_url(kwargs['translation'].language)
            request_custom.META['QUERY_STRING'] = ''
            cache_key = _generate_cache_header_key('', request_custom)
            cache.delete(cache_key)
    except Exception as e:
        logger.error(f'Error in page publish receiver: {e}')


signals.post_obj_operation.connect(delete_cache_key, PageAdmin)
```

**Comment by PeterW-LWL** [comment]
Is this still an issue in django-cms 3.11?

**Comment by fsbraun** [comment]
@PeterW-LWL I have not checked this, but I am not aware of a fix. Want to give it a go?

**Comment by PeterW-LWL** [comment]
> @PeterW-LWL I have not checked this, but I am not aware of a fix. Want to give it a go?

No. I was asking because we recently upgraded to 3.11 (from 3.7) and use a thirdparty library that solves this issue. But I wanted to make sure if we still need to depend on that library.  
But it looks like we do.

**Comment by PeterW-LWL** [comment]
https://gitlab.com/what-digital/djangocms-helpers

**Comment by Bhargav129** [comment]
is this bug is open? if it open i will try to fix.
